### DISCUSSION TITLE
Next.js 14: Partial Prerendering

### DISCUSSION THREAD

**Discussion by strvbkzhn** [vercel/next.js#58753] [Help]
### Summary

The documentation mentions that Partial Prerendering isn't functional for client-side navigation currently. Does this imply that dynamic content will be converted to static? Additionally, is there an estimated timeframe for when Partial Prerendering will be compatible with client-side navigation?

**Comment by christopher-caldwell** [comment]
I'm also curious about this. I don't exactly know what client side navigation means. Is that a `<Link />` component from next, or just the router.push?

My experience has been that everything works as expected with the suspense boundaries, but navigating with the router.push does not re-trigger Suspense. So it looks like nothing is happening for the duration of the request, then the content abruptly changes. 

The best example I can give is a product page with recommended for you similar products. You would load 4 at a time, with a server side pagination using query params. When someone paginates client side, the query param is added with a navigational push or through a Next Link. This works, and the new data is shown, but it does not re-trigger the suspense fallback, making it difficult to show a loading state. 

I am happy to make a reproduction, I'm just not sure if this is intentional right now with the development timeline. 

Also like to add that adding a key to the Suspense makes it re-trigger in development, but has no effect when hosted on Vercel.
